ykdl/util/jsengine.py

```diff
@@ -171,7 +171,13 @@ class ProgramError(Exception):
 
 # macOS: built-in JavaScriptCore
 if platform.system() == 'Darwin':
-    external_interpreter = '/System/Library/Frameworks/JavaScriptCore.framework/Versions/A/Resources/jsc'
+    # jsc lives on a new path since macOS Catalina
+    jsc_paths = ['/System/Library/Frameworks/JavaScriptCore.framework/Versions/A/Resources/jsc',
```

Comment:
If the path is not fixed, how about direct use `external_interpreter = which('jsc')`, is this working?

Reply:
Sadly I do not think we can use `external_interpreter = which('jsc')` because `jsc` is not in PATH by default